src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/csip/CSIPService.kt

```diff
@@ -63,6 +64,41 @@ class CSIPService(
     }
   }
 
+  suspend fun updateCSIPReport(csipEvent: CSIPEvent) {
+    val dpsCsipReportId = csipEvent.additionalInformation.recordUuid
+    val offenderNo = requireNotNull(csipEvent.personReference.findNomsNumber())
+    val telemetryMap = mutableMapOf(
+      "dpsCSIPReportId" to dpsCsipReportId,
+      "offenderNo" to offenderNo,
+    )
+
+    runCatching {
+      val mapping = mappingApiService.getOrNullByDpsId(dpsCsipReportId)
+        ?: throw IllegalStateException("Tried to update an csip that has never been created")
+      telemetryMap["nomisCSIPReportId"] = mapping.nomisCSIPReportId.toString()
+
+      val dpsCsipRecord = dpsApiService.getCsipReport(dpsCsipReportId)
+      nomisApiService.upsertCsipReport(dpsCsipRecord.toNomisUpsertRequest(mapping.nomisCSIPReportId))
+        .let { nomisCsip ->
+          CSIPFullMappingDto(
+            dpsCSIPReportId = dpsCsipReportId,
+            nomisCSIPReportId = nomisCsip.nomisCSIPReportId,
+            mappingType = CSIPFullMappingDto.MappingType.DPS_CREATED,
+            attendeeMappings = listOf(),
+            factorMappings = listOf(),
+            interviewMappings = listOf(),
+            planMappings = listOf(),
+            reviewMappings = listOf(),
+          )
+        }
+      // TODO Check and update child mappings
```

Comment:
There is still additional work to do here - as new child mappings may have been created - will do in a separate PR